src/test/java/io/appium/java_client/service/local/ServerBuilderTest.java

```diff
@@ -312,4 +316,45 @@ public void checkAbilityToStartServiceWithLogFileUsingShortFlag() {
         service.start();
         assertTrue(testLogFile.exists());
     }
+
+    @Test
+    public void checkAbilityToStartServiceUsingValidBasePathWithMultiplePathParams() {
+        String baseUrl = String.format("http://%s:%d/", BROADCAST_IP_ADDRESS, DEFAULT_APPIUM_PORT);
+        String basePath = "wd/hub";
+        service = new AppiumServiceBuilder().withArgument(BASEPATH, basePath).build();
+        service.start();
+        assertTrue(service.isRunning());
+        assertEquals(baseUrl + basePath + "/", service.getUrl().toString());
+    }
+
+    @Test
+    public void checkAbilityToStartServiceUsingValidBasePathWithSinglePathParams() {
+        String baseUrl = String.format("http://%s:%d/", BROADCAST_IP_ADDRESS, DEFAULT_APPIUM_PORT);
+        String basePath = "/wd/";
+        service = new AppiumServiceBuilder().withArgument(BASEPATH, basePath).build();
+        service.start();
+        assertTrue(service.isRunning());
+        assertEquals(baseUrl + basePath.substring(1), service.getUrl().toString());
+    }
+
+    @Test
+    public void checkAbilityToValidateBasePathForEmptyBasePath() {
+        assertThrows(IllegalArgumentException.class, () -> {
```

Comment:
FYI: you could add expected exception signature to Test annotations: https://www.guru99.com/junit-exception-test.html

Reply:
Noted with thanks.